test/elm/library/library-test.ts

```diff
@@ -213,3 +213,40 @@ describe('Using CommonLib and CommonLib2 with namespace support', () => {
     should.exist(this.common2LocalIdObject[sortUsingFunctionLocalId]);
   });
 });
+
+describe('CommonLib3', () => {
+  beforeEach(function () {
+    setup(this, data, [p1, p2], {}, {}, new Repository(data));
+  });
+  it('should be able to execute an expensive expression', async function () {
+    const startTime = Date.now();
+    this.results = await this.executor
+      .withLibrary(this.lib)
+      .exec_patient_context(this.patientSource);
+    const endTime = Date.now();
+    const executionTime = endTime - startTime;
+    executionTime.should.be.lessThan(250);
+    this.results.patientResults['1'].ExpensiveStatementRef.length.should.equal(1);
+  });
+});
+
+describe('Using CommonLib3', () => {
+  beforeEach(function () {
+    setup(this, data, [p1, p2], {}, {}, new Repository(data));
+  });
+
+  it('should execute expression from included library at similar cost', async function () {
+    const startTime = Date.now();
+    this.results = await this.executor
+      .withLibrary(this.lib)
+      .exec_patient_context(this.patientSource);
+    const endTime = Date.now();
+    const executionTime = endTime - startTime;
+    executionTime.should.be.lessThan(1000);
```

Comment:
I suggest re-working this test to not be based off of execution time as this can vary wildly based on the hardware it is run on. This also adds unnecessary execution time for testing.

Perhaps the expression run more than once could simply include a `Message` call and the unit test could create a message listener and attach before executing. After execution the number of times the message listener is hit could be checked.

Reply:
I just pulled in the test cases that were in #329 .  I'm not sure how to really tests this seeing how the underlying issue is really related to how the args are executed to prevent them from executing the same expressions.

Reply:
I agree with @hossenlopp that this will probably cause more trouble than it's worth.

Could we use a [Sinon spy](https://sinonjs.org/releases/v19/spies/) to spy on the `exec` method of the expression that should be cached and then assert that it was only called once?

Reply:
Oh! I somehow missed @hossenlopp's 2nd paragraph where he suggests using `Message` to track how many times the expression is executed. Yes, I think that would work too!

Reply:
I removed the timing based tests and added a test using sinon to spy on the exec method for the expression.